### PR TITLE
docs: adds/corrects EE changes, merges to OSS

### PR DESCRIPTION
**Pull request by wes-turner**
A few separate changes:
* A couple EE features/improvements didn't make the EE release notes. This adds them.
* A couple EE features/improvements were added tothe wrong version in the EE release notes. This moves them to the one they were actually released as a part of.
* Moves EE-only release notes to OSS and tags them (*Enterprise edition only*).
  - starting with 0.26.5
* Tags EE-only release notes that were already in OSS with an explicit (*Enterprise edition only*).
  - starting with 0.26.5

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/release-notes.rst [hunk 1]
```diff
@@ -99,12 +125,12 @@ Version 0.26.5
...
-   environment variable called ``DEX_TOKEN`` in task containers.
+-  Authentication: *(Enterprise edition only)* Users can now provide a Pachyderm address in the
+   master config under ``integrations.pachyderm.address``. This address will be added as an
+   environment variable called ``PACHD_ADDRESS`` in task containers. The OIDC raw ID token will also
+   be available as an environment variable called ``DEX_TOKEN`` in task containers.
 
--  Authentication: In the enterprise edition, add synchronization of OIDC user group memberships
+-  Authentication: *(Enterprise edition only)* Add synchronization of OIDC user group memberships
    with existing groups. Configure by setting ``oidc.groups_attribute_name`` in the master config to
    the string value of the authenticator's claim name for groups.
```
Comment (tara-det-ai) [inline docs/release-notes.rst:135]: in 0.26.5 wasn't it set as `oidc.groups_claim_name` and then it switched to `oidc.groups_attribute_name` later (perhaps in 0.26.7?)

File: docs/release-notes.rst [hunk 2]
```diff
@@ -99,12 +125,12 @@ Version 0.26.5
...
-   ``PACHD_ADDRESS`` in task containers. The OIDC raw ID token will also be available as an
-   environment variable called ``DEX_TOKEN`` in task containers.
+-  Authentication: *(Enterprise edition only)* Users can now provide a Pachyderm address in the
+   master config under ``integrations.pachyderm.address``. This address will be added as an
+   environment variable called ``PACHD_ADDRESS`` in task containers. The OIDC raw ID token will also
+   be available as an environment variable called ``DEX_TOKEN`` in task containers.
 
--  Authentication: In the enterprise edition, add synchronization of OIDC user group memberships
+-  Authentication: *(Enterprise edition only)* Add synchronization of OIDC user group memberships
    with existing groups. Configure by setting ``oidc.groups_attribute_name`` in the master config to
```
Comment (salonig23) [inline docs/release-notes.rst:134]: We have ``oidc.groups_attribute_name`` and ``oidc.groups_claim_name`` in line 95. One of these needs to be changed to make it uniform

Comment (wes-turner) [reply docs/release-notes.rst]: Fixed!

File: docs/release-notes.rst (hunk 1 above)
Comment (wes-turner) [reply docs/release-notes.rst:135]: Fixed!

**Review by salonig23 — COMMENTED** [review]
LGTM!

**Pull request merged**